Source/RuntimeMeshComponent/Public/RuntimeMeshGenericVertex.h

```diff
@@ -1007,214 +949,288 @@ DECLARE_RUNTIME_MESH_VERTEX(FRuntimeMeshVertexNoPositionDualUVHiPrecisionNormals
+        }
+    };
+    
+    
+    
+    //////////////////////////////////////////////////////////////////////////
+    // Vertex Structure Helper
+    //////////////////////////////////////////////////////////////////////////
+    template<bool WantsPosition, bool WantsNormal, bool WantsTangent, bool WantsColor, int32 NumWantedUVChannels,
+    ERuntimeMeshVertexTangentBasisType NormalTangentType, ERuntimeMeshVertexUVType UVType>
+    static RuntimeMeshVertexStructure CreateVertexStructure(const FVertexBuffer& VertexBuffer)
+    {
+        typedef FRuntimeMeshVertex<WantsPosition, WantsNormal, WantsTangent, WantsColor, NumWantedUVChannels, NormalTangentType, UVType> RuntimeVertexType;
+        
+        RuntimeMeshVertexStructure VertexStructure;
+        
+        // Add Position component if necessary
+        FRuntimeMeshPositionComponentVertexStructure<RuntimeVertexType, WantsPosition>::AddComponent(VertexBuffer, VertexStructure);
+        
+        // Add normal and tangent components if necessary
+        FRuntimeMeshNormalTangentComponentVertexStructure<RuntimeVertexType, WantsNormal, WantsTangent, NormalTangentType>::AddComponent(VertexBuffer, VertexStructure);
+        
+        // Add color component if necessary
+        FRuntimeMeshColorComponentVertexStructure<RuntimeVertexType, WantsColor>::AddComponent(VertexBuffer, VertexStructure);
+        
+        // Add all texture channels
+        FRuntimeMeshTextureChannelsVertexStructure<RuntimeVertexType, NumWantedUVChannels, UVType>::AddChannels(VertexBuffer, VertexStructure);
+        
+        return VertexStructure;
+    }
+};
 
-	//////////////////////////////////////////////////////////////////////////
-	// Normal/Tangent Components
-	//////////////////////////////////////////////////////////////////////////
-	template<typename RuntimeVertexType, bool WantsNormal, bool WantsTangent, ERuntimeMeshVertexTangentBasisType NormalTangentType>
-	struct FRuntimeMeshNormalTangentComponentVertexStructure
-	{
-		static void AddComponent(const FVertexBuffer& VertexBuffer, RuntimeMeshVertexStructure& VertexStructure)
-		{
-			VertexStructure.TangentBasisComponents[1] = RUNTIMEMESH_VERTEXCOMPONENT(VertexBuffer, RuntimeVertexType, Normal,
-				FRuntimeMeshVertexTangentTypeSelector<NormalTangentType>::VertexElementType);
-			VertexStructure.TangentBasisComponents[0] = RUNTIMEMESH_VERTEXCOMPONENT(VertexBuffer, RuntimeVertexType, Tangent,
-				FRuntimeMeshVertexTangentTypeSelector<NormalTangentType>::VertexElementType);
-		}
-	};
+// These need to be declared after FRuntimemeshVertexStructureHelper and RuntimeMeshVertexStructure to fix circular dependencies between the two
+template<bool WantsPosition, bool WantsNormal, bool WantsTangent, bool WantsColor, int32 NumWantedUVChannels,
+ERuntimeMeshVertexTangentBasisType NormalTangentType, ERuntimeMeshVertexUVType UVType>
+RuntimeMeshVertexStructure
```

Comment:
Hmm, just realized that this might need an inline since it's outside of the declaration.